ncm-network/src/main/perl/nmstate.pm

```diff
@@ -231,6 +231,34 @@ sub nmstate_file_dump
     }
 }
 
+# given ip/netmask address return cidr mask length
+sub get_masklen {
+    my ($self, $ip_netmask) = @_;
+    $self->debug(1, "Converting $ip_netmask to cidr notation");
+    my $ip = NetAddr::IP->new($ip_netmask);
+    return $ip->masklen;
+}
+
+# generate dummy interface config (aka loopback) for all service address vip
+# return hashrefs needed by nmstate interface config
+sub generate_vip_config {
+    my ($self, $iface_name, $vip) = @_;
+    my %dummy_iface;
+    my $ip_list = {};
+    my $netmask = $vip->{netmask} || "255.255.255.255";
+    my $ip = $vip->{ip};
+    $ip_list->{ip} = $ip;
+    $ip_list->{'prefix-length'} = get_masklen($self, "$ip/$netmask");
+    $dummy_iface{name} = $iface_name;
+    $dummy_iface{'profile-name'} = $iface_name;
+    $dummy_iface{type} = "dummy";
+    $dummy_iface{state} = "up";
+    $dummy_iface{ipv4}->{enabled} = "true";
```

Comment:
if you want a yaml boolean out of this, use the yaml constant for true. what does this geneate? the string true incl quotes, that evaluates to true by accident?